src/libultra_code/contpfs.c

```diff
@@ -0,0 +1,318 @@
+#include <ultra64.h>
+#include <global.h>
+
+extern __OSInode __osPfsInodeCache; // bss
+s32 	  __osPfsInodeCacheChannel = -1;
+u8		  __osPfsInodeCacheBank = 250;
+
+u16 __osSumcalc(u8* ptr, s32 length) {
+    s32 i;
+    u32 sum = 0;
+    u8* temp;
+
+    temp = ptr;
+    for (i = 0; i < length; i++) {
+        sum += *temp++;
+    }
+    return sum & 0xFFFF;
```

Comment:
The return type is already u16 so this mask is probably not needed?
```suggestion
    return sum;
```

Reply:
The mask is needed for this